hooks/react-client/useDetectOS.ts

```diff
@@ -4,33 +4,39 @@ import { useEffect, useState } from 'react';
 
 import type { UserOS } from '@/types/userOS';
 import { detectOS } from '@/util/detectOS';
+import { getArchitecture } from '@/util/getArchitecture';
 import { getBitness } from '@/util/getBitness';
 
 type UserOSState = {
   os: UserOS;
   bitness: number;
+  architecture: string;
 };
 
 const useDetectOS = () => {
   const [userOSState, setUserOSState] = useState<UserOSState>({
     os: 'OTHER',
     bitness: 86,
+    architecture: 'ARM',
   });
 
   useEffect(() => {
-    getBitness().then(bitness => {
-      const userAgent = navigator?.userAgent;
-
-      setUserOSState({
-        os: detectOS(),
-        bitness:
-          bitness === '64' ||
-          userAgent?.includes('WOW64') ||
-          userAgent?.includes('Win64')
-            ? 64
-            : 86,
-      });
-    });
+    Promise.all([getBitness(), getArchitecture()]).then(
+      ([bitness, architecture]) => {
+        const userAgent: string | undefined = navigator?.userAgent;
```

Comment:
```suggestion
        const userAgent: string | undefined = typeof navigator === 'object' && navigator.userAgent || '';
```

Reply:
Prevents exceptions if navigator would be ever not defined (the variable) and prevents the need of using `userAgent?.`